### DISCUSSION TITLE
GSI SK is always type string

### DISCUSSION THREAD

**Discussion by kinesra75** [sensedeep/dynamodb-onetable#211] [Q&A] [answered]
Hi all,

I try to add a GSI with a sort key type number.

Here is the pseudo  code :

```
indexes: {
        primary: { hash: 'pk', sort: 'sk' },
        gs1: { hash: 'entity', sort: '_createdAt', follow: true }
},
 models: {
        Enterprise: {
            pk: { type: String, value: 'ent#${id}' },
            sk: { type: String, value: 'ent#${id}' },
            id: { type: String, uuid: 'ulid' },
            status: { type: String, default: 'active' },
            entity: { type: String, default: 'enterprise', required: true },
            name: { type: String, required: true },
            _createdAt: { type: Number, default: Date.now() },
        },
}
```
But one the table is created, the gs1 sk is alway type string.

![image](https://user-images.githubusercontent.com/7263921/143485139-d4a61bc2-c4b6-42c7-9562-b12df8351312.png)

What I do wrong ?

Thanks in advance :)

**Comment by mobsense** [comment, marked as answer]
Can't tell easily without reproducing this -- could be a bug. Can you create an issue for this and create a test sample as per the issue template. Then we can try to reproduce and solve.

**Reply by mobsense** [reply]
Actually have some progress on this now.

CreateTable was hard coded for strings. We'll create a fix for this. It will use the model attribute type as you point out.

Note: you can use timestamps to create the 'createdAt' automatically for you. Set Table params.timestamps to true.

**Reply by kinesra75** [reply]
Hi,

Thanks a lot for your reply. It's a great project it save us a lot of time ! So I'll wait for the fix. Do you want me to create an issue or that's fine ?

Thanks

**Reply by mobsense** [reply]
No issue needed. It worked out fine. We'll aggregate the fix in the next patch release. Push if you need it sooner.
